pkg/tests/ossm/ior_test.go

```diff
@@ -0,0 +1,321 @@
+			}
+
+			t.LogStep("Check whether the IOR creates Routes for hosts specified in the Gateway")
+			createSimpleGateway(t)
+			checkSimpleGateway(t)
+		})
+
+		t.NewSubTest("check routes that are not deleted during v2.3 to v2.4 upgrade").Run(func(t test.TestHelper) {
+			t.Cleanup(func() {
+				oc.RecreateNamespace(t, meshNamespace)
+				setupDefaultSMCP(t, meshNamespace)
+			})
+
+			t.LogStepf("Delete and recreate namespace %s", meshNamespace)
+			oc.RecreateNamespace(t, meshNamespace)
+
+			t.LogStep("Deploy SMCP v2.3")
+			setupV23SMCP(t, meshNamespace, meshName)
+
+			t.LogStep("Check whether IOR creates Routes for hosts specified in the Gateway")
+			createSimpleGateway(t)
+			checkSimpleGateway(t)
+
+			t.LogStepf("Record the Route before the upgrade")
+			before := getRoutes(t, meshNamespace)
+
+			t.LogStep("Upgrade SMCP to v2.4")
+			updateToV24SMCP(t, meshNamespace, meshName)
+
+			t.LogStep("Check the Routes existed afther the upgrade")
+			checkSimpleGateway(t)
+			after := getRoutes(t, meshNamespace)
+
+			t.LogStep("Check the Routes unchanged afther the upgrade")
+			if before[0].Metadata.ResourceVersion != after[0].Metadata.ResourceVersion {
+				t.Fatal("Expect the route to be unchanged, but it is changed after the upgrade")
+			} else {
+				t.LogSuccess("Got the same resourceVersion before and after the upgrade")
+			}
+		})
+
+		t.NewSubTest("check IOR does not delete routes after deleting Istio pod").Run(func(t test.TestHelper) {
+			total := 3
+			nsNames := []string{}
+			gateways := []string{}
+
+			t.Cleanup(func() {
+				if env.GetSMCPVersion().GreaterThanOrEqual(version.SMCP_2_4) {
+					removeIORCustomSetting(t, meshNamespace, meshName)
+				}
+
+				oc.DeleteNamespace(t, nsNames...)
+				oc.ApplyString(t, meshNamespace, GetSMMRTemplate())
+				oc.WaitSMMRReady(t, meshNamespace)
+			})
+
+			t.LogStep("Ensure the IOR enabled")
+			if env.GetSMCPVersion().GreaterThanOrEqual(version.SMCP_2_4) {
+				enableIOR(t, meshNamespace, meshName)
+			}
+
+			t.LogStepf("Create %d Gateways and they are in their own Namespace", total)
+			for i := 0; i < total; i++ {
+				ns := fmt.Sprintf("ns-%d", i)
+				nsNames = append(nsNames, ns)
+				gateways = append(gateways, generateGateway(fmt.Sprintf("gw-%d", i), ns, fmt.Sprintf("www-%d.test.ocp", i)))
+			}
+			oc.CreateNamespace(t, nsNames...)
+			oc.ApplyString(t, "", gateways...)
+
+			t.LogStepf("Update SMMR to include %d Namespaces", total)
+			start := time.Now()
+			oc.ApplyString(t, meshNamespace, fmt.Sprintf(`
+apiVersion: maistra.io/v1
+kind: ServiceMeshMemberRoll
+metadata:
+  name: default
+spec:
+  members:
+  - bookinfo
+  - foo
+  - bar
+  - legacy
+  - %s
+  `, strings.Join(nsNames, "\n  - ")))
+
+			var end time.Time
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				routes := getRoutes(t, meshNamespace)
+				if len(routes) == total {
+					end = time.Now()
+					t.LogSuccessf("Found all %d Routes in %.2f", total, end.Sub(start).Seconds())
```

Comment:
No need to include the time here, as it just pollutes the test code. The retry mechanism already prints the total time needed for the check to succeed:
```
ior_test.go:179:    SUCCESS: Found all 3 Routes in 14.90
ior_test.go:175:    --- Attempt 14/60 successful; total time: 14.36s
```

Reply:
Removed the timing.